src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -8,16 +8,37 @@ namespace Microsoft.AspNetCore.Server.Kestrel
 {
     public class KestrelServerOptions
     {
+        private int? _maxInputBufferLength = 1024 * 1024;
```

Comment:
Did you base this on a limit on the limit of another server? It might be nice to add a non-doc comment explaining how we chose the default.